packages/ra-ui-materialui/src/list/List.spec.js

```diff
@@ -71,77 +75,54 @@ describe('<List />', () => {
         expect(queryAllByText('datagrid')).toHaveLength(1);
     });
 
-    const defaultListProps = {
-        basePath: '/foo',
-        ids: [],
-        data: {},
-        hasCreate: false,
-        hasEdit: false,
-        hasList: false,
-        hasShow: false,
-        location: {},
-        match: {},
-        resource: 'foo',
-        total: 0,
-    };
-
-    const defaultStateForList = {
-        admin: {
-            resources: {
-                foo: {
-                    list: {
-                        ids: [],
-                        params: {},
-                        selectedIds: [],
-                        total: 0,
-                        cachedRequests: {},
-                    },
-                },
-            },
-        },
-    };
-
     it('should display aside component', () => {
         const Dummy = () => <div />;
         const Aside = () => <div id="aside">Hello</div>;
         const { queryAllByText } = renderWithRedux(
             <ThemeProvider theme={theme}>
-                <List {...defaultListProps} aside={<Aside />}>
+                <List {...defaultProps} aside={<Aside />}>
                     <Dummy />
                 </List>
-            </ThemeProvider>,
-            defaultStateForList
+            </ThemeProvider>
         );
         expect(queryAllByText('Hello')).toHaveLength(1);
     });
 
-    it('should render an invite when the list is empty', () => {
+    it('should render an invite when the list is empty', async () => {
         const Dummy = () => <div />;
+        const dataProvider = {
+            getList: jest.fn(() => Promise.resolve({ data: [], total: 0 })),
+        };
         const { queryAllByText } = renderWithRedux(
             <ThemeProvider theme={theme}>
-                <ListView {...defaultProps} total={0} hasCreate loaded>
-                    <Dummy />
-                </ListView>
-            </ThemeProvider>
+                <DataProviderContext.Provider value={dataProvider}>
+                    <List {...defaultProps}>
+                        <Dummy />
+                    </List>
+                </DataProviderContext.Provider>
+            </ThemeProvider>,
+            defaultStateForList
         );
-        expect(queryAllByText('resources.post.empty')).toHaveLength(1);
+        await wait();
+        expect(queryAllByText('resources.posts.empty')).toHaveLength(1);
```

Comment:
You should put the expect inside the wait
```suggestion
        await wait(() => {
            expect(queryAllByText('resources.posts.empty')).toHaveLength(1);
        });
```